File: 2023-2024/sourcecode/mod1/lab2/22SelectionBasicsJava/app/src/main/java/com/example/a22_selectionbasics_java/NumberItemDetailsLookup.java
package com.example.a22_selectionbasics_java;

import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.selection.ItemDetailsLookup;
import androidx.recyclerview.widget.RecyclerView;

public class NumberItemDetailsLookup extends ItemDetailsLookup<Long> {

    private final RecyclerView recyclerView;

    public NumberItemDetailsLookup(RecyclerView recyclerView) {
        this.recyclerView = recyclerView;
    }

    public ItemDetails<Long> getItemDetails(@NonNull MotionEvent e) {
        View view = recyclerView.findChildViewUnder(e.getX(), e.getY());
        return ((NumberListViewHolder)recyclerView
                .getChildViewHolder(view)).getItemDetails();
    }
}

File: 2023-2024/sourcecode/mod1a/lab2/21SelectorBasicsKotlin/app/src/main/java/com/example/a21_selectorbasics_kotlin/NumberItemDetailsLookup.kt
package com.example.a21_selectorbasics_kotlin

import android.view.MotionEvent
import androidx.recyclerview.selection.ItemDetailsLookup
import androidx.recyclerview.widget.RecyclerView

class NumberItemDetailsLookup(private val recyclerView: RecyclerView) :
    ItemDetailsLookup<Long>() {
    override fun getItemDetails(event: MotionEvent): ItemDetails<Long>?{
        val view = recyclerView.findChildViewUnder(event.x, event.y)
        return (view?.let { recyclerView.getChildViewHolder(it) } as NumberListViewHolder).getItemDetails()
    }
}